### DISCUSSION TITLE
Excel interface to run modelx models without Python installation

### DISCUSSION THREAD

**Discussion by AB-Athene** [fumitoh/modelx#149] [Show and tell]
I found a nice and flexible way to run `modelx `models without Python installation.
I use Excel interface, which specifies models to run together with the parameters to use, when running them. Excel interface has a number of advantages:
- Extremely flexible GUI with color coding, input validations, separate tabs, ...
- Users easily can help developing the interface
- Simple tasks like folder creation, checks can be facilitated by VBA - again people without Python skills can contribute and customize for their needs, keeping the Python code common

There is a very simple integration between Excel interface and Python: Excel calls Python executable together with JSON string of parameters using Shell command. The executable is created using `pyinstaller`.
The models are run in parallel based on the number of parallel processes specified in the interface. 

Attached example illustrates it. Instructions:
1. In Python Console create new savings models from `lifelib`: `lifelib.create("savings", path)`
2. Specify path to the models in table in Excel file on the right
3. Specify path to Python.exe, model_run_test.py and Output folder on top.
4. Click "Run Models Python" button
This will run model_run_test.py script relying on your local Python and libraries

You can instead create an executable by running `pyinstaller model_run_test.spec` in the Terminal and specify the path to the executable as App path in the Excel file. "Run Models exe" will run the models using the exe file.

[model_run_test.zip](https://github.com/user-attachments/files/16716794/model_run_test.zip)

**Comment by fumitoh** [comment]
Thanks a lot of sharing. I'll consider including the example in lifelib.

**Reply by AB-Athene** [reply]
Note that 
- I don't use it with `lifelib` models, but custom models built on `modelx` engine.
- I use it in a combination with zipped models, which makes it very easy to distribute the model to the users, who can run it without updating the executable or the Excel interface. Essentially, sharing mx.zip file becomes similar to sharing Excel files.
